float32 thrust
float32 heading